harmony/harmonydb/sql/20241028-padding.sql

```diff
@@ -0,0 +1,171 @@
+CREATE TABLE deal_index_tracker (
```

Comment:
Why do we need this instead of looking e.g. at the _initial_pieces tables and the piece_index there?

Reply:
I wasn't a big fan of this approach while I was writing it. I was trying to find some way to avoid state but could not think of a reliable way which did not involve some really long function. Here is the problem:
1. Typically we can easily identify the piece with piece CID and Size.
2. Problem starts when we have the same piece in more than 1 location in the same sector. How do I reliably connect a piece to a deal. We don't even need to index it later but we still need to know about the offset. This was a reason I went with a stateful solution. 

Another approach I can think of would be to add an index_column to market_mk12_deal_pipeline table and populate it during the transfer functions from within the SQL. It seems cheaper than the current solution.

If you have any insight which avoids any of this, I can really use it.

Reply:
From Slack:
We are not going to allow same piece more than once in a sector. This removes the stateful tracking to find unique pieces and their offset in sector.